### ISSUE TITLE
Jenkins slave should use a known good/local maven mirror

### ISSUE THREAD

**Issue by jfchevrette** [openshiftio/openshift.io#2457]
As I was testing #2368 today I noticed downloading the dependencies was still painfully slow. It took ~10 minutes to download them even on local/ephemeral storage.

I had a look at the settings.xml we provide and it doesn't contain anything wrt. mirrors. Maven probably picks a mirror at random (?) and we seem to be hitting a bad mirror most of the time.

I had a look at Che's settings.xml (where my build are crazy fast) and noticed some differences in the repo profiles.

Here are the changes I made to my own tenant's settings.xml:
https://gist.github.com/jfchevrette/73b8faa9edeac742d7884f6c2f365e70/revisions

Upon doing that, build times (on local storage) went from 15-20 minutes to <5 minutes consistently.

We should make sure we use a good/local mirror.

**Comment by jaseemabid** [comment]
@jfchevrette Can you make this change on the tenant Jenkins as well?

**Comment by jfchevrette** [comment]
@jaseemabid I'd like someone from the build team to validate the config first. I merely copy/pasted configs from other settings.xml and it happened to work great for me. We should make sure we understand what this does and if hardcoding a repo like that is the way to go.

@aslakknutsen how can we update secret jenkins-maven-settings on all existing tenants as change the default for new tenants?

**Comment by chmouel** [comment]
@jaseemabid @rupalibehera @sthaha

**Comment by jfchevrette** [comment]
@aslakknutsen sure, but where does the jenkins-maven-settings come from when we init a tenant?

**Comment by aslakknutsen** [comment]
https://github.com/fabric8-services/fabric8-tenant-jenkins/blob/master/apps/jenkins/src/main/fabric8/jenkins-maven-settings-secret.yml

**Comment by pradeepto** [comment]
@jaseemabid Are you working on this?If so, can you assign yourself?

**Comment by pradeepto** [comment]
@chmouel Should this part of the sprint 146?

**Comment by jaseemabid** [comment]
@hrishin was looking into this I believe?

**Comment by hrishin** [comment]
> I'd like someone from the build team to validate the config first. I merely copy/pasted configs from other settings.xml and it happened to work great for me. 

We have added the same maven repository URL's to an existing profile that is being used. 
https://gist.github.com/hrishin/6e45d64889c6e732f9cb51595442b43b

> We should make sure we understand what this does

<[repository](http://maven.apache.org/ref/3.5.2/maven-settings/settings.html)> contains the information needed for establishing connections with remote repository.


> if hardcoding a repo like that is the way to go.

As this is a confinmap (secretes), relatively it could be feasible to update it later on.

**Comment by kbsingh** [comment]
ping @chmouel @pradeepto can we get this moving and deployed ? the condition this fix's is being reported by the users being onboarded at this time

**Comment by chmouel** [comment]
I have already asked that same question to you this morning : 

![image](https://user-images.githubusercontent.com/98980/37360603-0eb2e278-26f1-11e8-9612-66c5d0c79104.png)

We have the fix ready but to deploy it there is something to do on the SD side (see conversation above with aslak)  to a cluster wide update,

**Comment by kbsingh** [comment]
SD does not own any content, the build team is responsible for getting this shipped out guys. Do you need to speak with the platform team around the tenant space perhaps ?

**Comment by hrishin** [comment]
Merged the relevant PR https://github.com/fabric8-services/fabric8-tenant-jenkins/pull/74

@aslakknutsen could you please let us know how we can ship the new release of [fabric8-tenant-jenkins](https://github.com/fabric8-services/fabric8-tenant-jenkins) to prod?
I guess some CI job should update the [JENKINS_VERSION](https://github.com/fabric8-services/fabric8-tenant/blob/master/JENKINS_VERSION#L1) in fabric8-tanent repo

cc: @pbergene @jfchevrette @chmouel @rupalibehera

**Comment by hrishin** [comment]
### Status update

A new version of fabric8-tenant-jenkins is available: [v4.0.62](https://github.com/fabric8-services/fabric8-tenant/blob/master/JENKINS_VERSION#L1) in [fabric8-tenant](https://github.com/fabric8-services/fabric8-tenant) repo.

**Comment by rupalibehera** [comment]
@hrishin we are done from our side the changes would reflect to prod after tenant wide update which SD team will handle or someone who does the tenant wide update.

**Comment by kbsingh** [comment]
Can i request the team to then open a new issue, assign it to the platform team ( since they own tenant interface service ), mark it as a SEV2 to get a cluster upgrade done. When you have that issue, please link to it here. Once the work is done to get the cluster upgrades, please verify and close this issue as fixed, pushed.

**Comment by hrishin** [comment]
For VertX HTTP booster it took
```
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 02:04 min
[INFO] Finished at: 2018-03-15T03:34:44+00:00
[INFO] Final Memory: 44M/64M
[INFO] ------------------------------------------------------------------------
```
Jenkins Version: 4.0.62, Cluster: starter-us-east-2  
Jenkins logs: https://pastebin.com/DMrbtcQz

**Comment by pradeepto** [comment]
@hrishin So can we close this?

**Comment by hrishin** [comment]
@pradeepto its good to close now.
cc: @kbsingh @jfchevrette @rupalibehera
